integrated_channels/moodle/exporters/content_metadata.py

```diff
@@ -0,0 +1,77 @@
+"""
+Content metadata exporter for Moodle
+"""
+
+import time
+
+from logging import getLogger
+
+from integrated_channels.integrated_channel.exporters.content_metadata import ContentMetadataExporter
+
+LOGGER = getLogger(__name__)
+
+
+class MoodleContentMetadataExporter(ContentMetadataExporter):
+    """
+        Moodle implementation of ContentMetadataExporter.
+    """
+    DATA_TRANSFORM_MAPPING = {
+        'fullname': 'key',
+        'shortname': 'key',
+        'idnumber': 'number',
+        'summary': 'description',
+        'startdate': 'start',
+        'enddate': 'end',
+        'categoryid': 'categoryid',
+    }
+
+    LONG_STRING_LIMIT = 2000
+
+    def transform_description(self, content_metadata_item):
+        """
+        Return the course description and enrollment url as Moodle' syllabus body attribute.
+        This will display in the Syllabus tab in Moodle.
+        """
+        enrollment_url = content_metadata_item.get('enrollment_url', None)
+        base_description = "<a href={enrollment_url}>To edX Course Page</a><br />".format(
+            enrollment_url=enrollment_url)
+        full_description = content_metadata_item.get('full_description') or None
+        if full_description and len(full_description + enrollment_url) <= self.LONG_STRING_LIMIT:
+            description = "{base_description}{full_description}".format(
+                base_description=base_description,
+                full_description=full_description
+            )
+        else:
+            short_description = content_metadata_item.get(
+                'short_description', content_metadata_item.get('title', '')
+            )
+            description = "{base_description}{short_description}".format(
+                base_description=base_description, short_description=short_description
+            )
+
+        return description
+
+    def transform_categoryid(self, content_metadata_item):  # pylint: disable=unused-argument
+        """
+        Returns the Moodle category id. ID 1 is Miscellaneous and is the default/basic category.
+        Will other categories actually be used in the future? Unclear.
+        """
+        return 1
+
+    def transform_start(self, content_metadata_item):
+        """
+        Converts start from ISO date string to int (required for Moodle's "startdate" field)
+        """
+        start_date = content_metadata_item.get('start', None)
+        if start_date:
+            return time.mktime(time.strptime(start_date, '%Y-%m-%dT%H:%M:%SZ'))
+        return None
+
+    def transform_end(self, content_metadata_item):
+        """
+        Converts end from ISO date string to int (required for Moodle's "enddate" field)
+        """
+        end_date = content_metadata_item.get('end', None)
+        if end_date:
+            return time.mktime(time.strptime(end_date, '%Y-%m-%dT%H:%M:%SZ'))
+        return None
```

Comment:
I realized to truly visualize and grasp the transmitter output and thus the final format we'd pass to the API, I'd want some data so oopsie daisy I basically wrote the exporter already.